packages/backend/convex/subscriptions.ts

```diff
@@ -1,243 +1,104 @@
 import { getAuthUserId } from "@convex-dev/auth/server";
+import { Polar as PolarComponent } from "@convex-dev/polar";
 import { Polar } from "@polar-sh/sdk";
+import { WebhookSubscriptionCreatedPayload$inboundSchema } from "@polar-sh/sdk/models/components";
 import { v } from "convex/values";
-import { api, internal } from "./_generated/api";
-import {
-  action,
-  internalMutation,
-  internalQuery,
-  mutation,
-  query,
-} from "./_generated/server";
+import { sortBy } from "remeda";
+import { api, components } from "./_generated/api";
+import type { Id } from "./_generated/dataModel";
+import { action, internalMutation, mutation, query } from "./_generated/server";
 import { env } from "./env";
-import schema from "./schema";
 
-const createCheckout = async ({
-  customerEmail,
-  productPriceId,
-  successUrl,
-  subscriptionId,
-}: {
-  customerEmail: string;
-  productPriceId: string;
-  successUrl: string;
-  subscriptionId?: string;
-}) => {
-  const polar = new Polar({
-    server: "sandbox",
-    accessToken: env.POLAR_ACCESS_TOKEN,
-  });
-  const result = await polar.checkouts.create({
-    productPriceId,
-    successUrl,
-    customerEmail,
-    subscriptionId,
-  });
-  return result;
-};
-
-export const getPlanByKey = internalQuery({
-  args: {
-    key: schema.tables.plans.validator.fields.key,
-  },
-  handler: async (ctx, args) => {
-    return ctx.db
-      .query("plans")
-      .withIndex("key", (q) => q.eq("key", args.key))
-      .unique();
-  },
-});
+const polarComponent = new PolarComponent(components.polar);
+export { polarComponent as polar };
 
 export const getOnboardingCheckoutUrl = action({
+  args: {},
```

Comment:
was this in order to make something work? We have some issues with emitting declarations and explicit annotations or validators like this if this was fixing it'd be great to play with and document

Reply:
Oh this looks like just an artifact from toying with the function, I'll remove it.